### PR TITLE
SDK sample for .NET client library.

### PR DESCRIPTION
**Pull request by dpokluda**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by annelo-msft — CHANGES_REQUESTED** [review]
Would you please update the samples to follow the pattern illustrated in: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/template/Azure.Template/samples?  Salient characteristics of Azure SDK .NET Samples include:
- `samples` directory: `azure-sdk-for-net/tree/main/sdk/<service>/<library>/samples`
- `README.md` providing a TOC for all available samples under `<library>/samples`
- `Sample<i>_<scenario>.md` files describing the scenario the sample is illustrating under `<library>/samples`
- `test/samples` directory: `azure-sdk-for-net/tree/main/sdk/<service>/<library>/tests/samples`
- tests inheriting from TestFramework `SamplesBase` that implement all sample code under `<library>/tests/samples`
- [snippet regions](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/tests/Samples/MiniSecretClientSamples.HelloWorld.cs#L23) identifying snippets to show in README and Samples files.

A nice example of a service library that follows the .NET samples guidance is MetricsAdvisor: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/metricsadvisor/Azure.AI.MetricsAdvisor

**Review by lmazuel — APPROVED** [review]
Discussed offline we will accept the PR as-is and rework on it by end of Feb

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" Version="1.5.0" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj:13]: We'll need to remove the version number - it gets pulled in from [package.data.props](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L93)

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <PackageReference Include="Azure.IoT.DeviceUpdate" Version="1.0.0-beta.3" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj:14]:
```suggestion
    <PackageReference Include="Azure.IoT.DeviceUpdate" VersionOverride="1.0.0-beta.3" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <PackageReference Include="System.CommandLine.DragonFruit" Version="0.4.0-alpha.21617.1" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj:15]:
```suggestion
    <PackageReference Include="System.CommandLine.DragonFruit" VersionOverride="0.4.0-alpha.21617.1" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj (hunk 1 above)
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj:13]:
```suggestion
    <PackageReference Include="Azure.Identity" />
```

Comment (dpokluda) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj:13]: So just: `<PackageReference Include="Azure.Identity" />` ?

Comment (JoshLove-msft) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj:13]: Yes, I added this as a suggestion below that you can commit from the PR.

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <ExcludeFromCodeCoverage>true</ExcludeFromCodeCoverage>
+    <ImportRepoCommonSettings>true</ImportRepoCommonSettings>
+    <!-- Signal that samples are building in the repo as opposed to a standalone download from Samples Browser -->
+    <IsClientLibrary>true</IsClientLibrary>
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Directory.Build.props:6]:
```suggestion
    <IsClientLibrary>false</IsClientLibrary>
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <ExcludeFromCodeCoverage>true</ExcludeFromCodeCoverage>
+    <ImportRepoCommonSettings>true</ImportRepoCommonSettings>
+    <!-- Signal that samples are building in the repo as opposed to a standalone download from Samples Browser -->
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Directory.Build.props:5]:
```suggestion
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/GetUpdate/GetUpdate.csproj [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/GetUpdate/GetUpdate.csproj:14]:
```suggestion
    <PackageReference Include="Azure.Identity" VersionOverride="1.5.0" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeployUpdate/DeployUpdate.csproj:14]:
```suggestion
    <PackageReference Include="Azure.Identity" VersionOverride="1.5.0" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeleteUpdate/DeleteUpdate.csproj [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/DeleteUpdate/DeleteUpdate.csproj:14]:
```suggestion
    <PackageReference Include="Azure.Identity" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/EnumerateUpdates/EnumerateUpdates.csproj [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/EnumerateUpdates/EnumerateUpdates.csproj:14]:
```suggestion
    <PackageReference Include="Azure.Identity" VersionOverride="1.5.0" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/GetDevice/GetDevice.csproj [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/GetDevice/GetDevice.csproj:14]:
```suggestion
    <PackageReference Include="Azure.Identity" VersionOverride="1.5.0" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/ImportUpdate/ImportUpdate.csproj [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/ImportUpdate/ImportUpdate.csproj:14]:
```suggestion
    <PackageReference Include="Azure.Identity" VersionOverride="1.5.0" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/ImportUpdate/ImportUpdate.csproj [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Constant.cs" Link="Constant.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" VersionOverride="1.5.0" />
+    <PackageReference Include="Azure.IoT.DeviceUpdate" VersionOverride="1.0.0-beta.3" />
+    <PackageReference Include="Azure.Storage.Blobs" />
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/ImportUpdate/ImportUpdate.csproj:16]:
```suggestion
    <PackageReference Include="Azure.Storage.Blobs" VersionOverride="12.10.0" />
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Directory.Build.props [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <ExcludeFromCodeCoverage>true</ExcludeFromCodeCoverage>
+    <ImportRepoCommonSettings>true</ImportRepoCommonSettings>
+    <IsClientLibrary>false</IsClientLibrary>
+    <IsPackable>true</IsPackable>
+    <IsSample>true</IsSample>
```
Comment (JoshLove-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Directory.Build.props:7]:
```suggestion
    <IsSamplesProject>true</IsSamplesProject>
```

**Pull request merged**